soc/arm64/nxp_layerscape/ls1046a/soc.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <init.h>
+#include <soc.h>
+#include <arch/cpu.h>
+
+/**
+ * @brief Perform basic hardware initialization at boot.
+ *
+ * This needs to be run from the very beginning.
+ * So the init priority has to be 0 (zero).
+ *
+ * @return 0
+ */
+static int ls1046a_init(const struct device *arg)
+{
+	uint32_t key;
+
+	ARG_UNUSED(arg);
+
+	key = irq_lock();
+
+	/*todo: add soc init code here*/
+
+	irq_unlock(key);
```

Comment:
so, we can remove the lock as well given there is no code at the moment?

Reply:
Yes, currently we can remove it. thanks.

Jiafei.